File: twosum/s.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       unordered_map<int, int> m;
        int size = nums.size();
        for (int i = 0; i < size; i++) {
            int a = target - nums[i];
            if (m.find(a) != m.end()) {
                return {m[a], i};
            }
            else
                m[nums[i]] = i;
        }
        return {};
        // for (int i = 0; i < nums.size(); i++)
        // {
        //     for (int j = i + 1; j < nums.size(); j++)
        //     {
        //         if (nums[i] + nums[j] == target)
        //         {
        //             return {i, j};
        //         }
        //     }
        // }
        // return {};
    }
};

File: twosum/s.cs
using System.Collections;
using System.Collections.Generic;

namespace s1
{
    public class Solution
    {
        public int[] TwoSum(int[] nums, int target)
        {
            Dictionary<int, int> map = new Dictionary<int, int>();
            for (var i = 0; i < nums.Length; i++)
            {
                map[nums[i]] = i;
            }
            for (var i = 0; i < nums.Length; i++)
            {
                var r = target - nums[i];
                if (map.ContainsKey(r) && map[r] != i)
                {
                    return new int[] { i, map[r] };
                }
            }
            return null;

        }

        public int[] s2(int[] nums, int target)
        {
            var x = 0;
            var y = 0;
            for (; x < nums.Length; x++)
            {
                for (y = x + 1; y < nums.Length; y++)
                {
                    if (nums[x] + nums[y] == target)
                    {
                        return new int[] { x, y };
                    }
                }
            }
            return null;
        }
    }
}